Assets/MixedRealityToolkit.Tests/PlayModeTests/InputEventSystemTests.cs

```diff
@@ -0,0 +1,300 @@
+
+            Assert.AreEqual(inputSystem.EventHandlersByType[typeof(IMixedRealityBaseInputHandler)].Count, 1, "Input event system doesn't contain expected IMixedRealityBaseInputHandler handlers.");
+            CollectionAssert.Contains(inputSystem.EventHandlersByType[typeof(IMixedRealityBaseInputHandler)], handlerBasedListener, "Input event system doesn't contain IMixedRealityBaseInputHandler for new API handler.");
+
+            objectBasedListener.enabled = false;
+
+            // Make sure that disabling global listener doesn't remove the new API one.
+            Assert.AreEqual(inputSystem.EventHandlersByType.Count, 2, "Input event system contains unexpected event handlers.");
+
+            Assert.AreEqual(inputSystem.EventHandlersByType.Count, 2, "Input event system doesn't contain expected event handler types.");
+            CollectionAssert.Contains(inputSystem.EventHandlersByType.Keys, typeof(IMixedRealitySpeechHandler), "Input event system doesn't contain IMixedRealitySpeechHandler entry.");
+            CollectionAssert.Contains(inputSystem.EventHandlersByType.Keys, typeof(IMixedRealityBaseInputHandler), "Input event system doesn't contain IMixedRealityBaseInputHandler entry.");
+
+            Assert.AreEqual(inputSystem.EventHandlersByType[typeof(IMixedRealitySpeechHandler)].Count, 1, "Input event system doesn't contain expected IMixedRealitySpeechHandler handlers.");
+            CollectionAssert.Contains(inputSystem.EventHandlersByType[typeof(IMixedRealitySpeechHandler)], handlerBasedListener, "Input event system doesn't contain IMixedRealitySpeechHandler for new API handler.");
+
+            Assert.AreEqual(inputSystem.EventHandlersByType[typeof(IMixedRealityBaseInputHandler)].Count, 1, "Input event system doesn't contain expected IMixedRealityBaseInputHandler handlers.");
+            CollectionAssert.Contains(inputSystem.EventHandlersByType[typeof(IMixedRealityBaseInputHandler)], handlerBasedListener, "Input event system doesn't contain IMixedRealityBaseInputHandler for new API handler.");
+
+            handlerBasedListener.enabled = false;
+            Assert.AreEqual(inputSystem.EventHandlersByType.Count, 0, "Input event system contains unexpected event handlers.");
+
+            Object.Destroy(object1);
+            yield return null;
+        }
+
+        /// <summary>
+        /// </summary>
+        [UnityTest]
+        public IEnumerator Test02_TestHandlerBasedEventRegistration()
+        {
+            yield return SetupMrtkWithoutCursors();
+
+            IMixedRealityInputSystem iInputSystem = null;
+            MixedRealityServiceRegistry.TryGetService(out iInputSystem);
+
+            BaseEventSystem inputSystem = (BaseEventSystem)iInputSystem;
+
+            var object1 = new GameObject("Object");
+
+            // These 2 handlers are independent
+            var handlerBasedListener1 = object1.AddComponent<TestInputGlobalListenerHandlerBasedAllHandlers>();
```

Comment:
Nit: It might help with readability to have a comment about what specific handlers each of these components register, so that within the context of this test, you can be like:

#1  = pointer speech base
#2  = speech base

If you read the test from top to bottom you have your mindset now where you know that the diff is just pointer, so when disabling/adding stuff down below it's a bit easier to understand the expected deltas.